java/arcs/core/allocator/Allocator.kt

```diff
@@ -122,30 +115,29 @@ class Allocator private constructor(
         partitionMap[arcId] = partitions
 
         partitions.forEach { partition ->
-            val singletons = mapOf(
-                "arc" to arcId.toString().toReferencable(),
-                "host" to partition.arcHost.toReferencable()
+            val entity = EntityBase("EntityBase", SCHEMA)
+            entity.setSingletonValue("arc", arcId.toString())
+            entity.setSingletonValue("host", partition.arcHost)
+            entity.setCollectionValue(
+                "particles",
+                partition.particles.map { it.particleName }.toSet()
             )
-            val collections = mapOf(
-                "particles" to partition.particles.map {
-                    it.particleName.toReferencable()
-                }.toSet()
-            )
-            val entity = RawEntity(arcId.toString() + partition.arcHost, singletons, collections)
             collection.store(entity)
         }
     }
 
     /** Looks up [RawEntity]s representing [PlanPartition]s for a given [ArcId] */
-    private suspend fun entitiesForArc(arcId: ArcId): List<RawEntity> =
-        collection.fetchAll().filter { fieldAsString(it, "arc") == arcId.toString() }
+    private suspend fun entitiesForArc(arcId: ArcId): List<EntityBase> =
+        collection.fetchAll().filter { it.getSingletonValue("arc") == arcId.toString() }
```

Comment:
"arc" is repeated here and elsewhere, maybe extract this to a const val? ditto for "host" and "particles"

Reply:
It's basically a const val already isn't it? Is there a major advantage to using ARC over "arc"?